bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/InstallToolsHandler.java

```diff
@@ -83,6 +95,38 @@ protected IStatus run(IProgressMonitor monitor)
 		installToolsJob.schedule();
 
 	}
+	
+	private void copyOpenOcdRules()
+	{
+		if (Platform.getOS().equals(Platform.OS_LINUX))
+		{
+			if (!IDFUtil.getOpenOCDLocation().equalsIgnoreCase(StringUtil.EMPTY))
+			{
+				console.println(Messages.InstallToolsHandler_CopyingOpenOCDRules);
+				// Copy the rules to the idf
+				StringBuilder pathToRules = new StringBuilder();
+				pathToRules.append(IDFUtil.getOpenOCDLocation());
+				pathToRules.append("/../share/openocd/contrib/60-openocd.rules"); //$NON-NLS-1$
+				File rulesFile = new File(pathToRules.toString());
+				if (rulesFile.exists())
+				{
+					Path source = Paths.get(pathToRules.toString());
+					Path target = Paths.get("/etc/udev/rules.d/60-openocd.rules"); //$NON-NLS-1$
+					try
+					{
+						Files.copy(source, target, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Could you print the source to the destination copy statement with paths in the IDF Console? If something goes wrong while copying the file due to some permissions, the user could try to do it manually if knew what we are trying to do.

Reply:
Done